ReSharper.FSharp/src/FSharp.Psi/src/IFSharpElementFactory.cs

```diff
@@ -64,5 +65,13 @@ public interface IFSharpElementFactory
     ITypeParameterDeclarationList CreateTypeParameterOfTypeList(FSharpList<string> names);
 
     IAccessorsNamesClause CreateAccessorsNamesClause(bool withGetter, bool withSetter);
+
+    IFSharpFile CreateEmptyFile();
+    INamedModuleDeclaration CreateModule(string name);
+    INamespaceDeclaration CreateNamespace(string name);
+    INestedModuleDeclaration CreateNestedModule(string name);
+
+    IModuleMember CreateModuleMember(string source);
+    IFSharpTypeMemberDeclaration CreateTypeMemberSignature(string source);
```

Comment:
It seems it can also be used to create non-signature type members? If true, then maybe something like `CreateTypeMember` would be a better name.